### PR TITLE
Pass args to parser and handle absolute out path

### LINKED ISSUE #333: parser --out doesn't handle absolute path
```
~/repo/git-truck/parser$ npm start -- --path ~/repo/bong-maps/ --out ~/data.json

> parser@1.0.0 start
> npm run build && node ./dist "--path" "/home/hojelse/repo/bong-maps/" "--out" "/home/hojelse/data.json"


> parser@1.0.0 build
> tsc

✔ Found branch head [6ms]
✔ Commit tree parsed [379ms]
✔ Commit tree hydrated [3.23s]
✔ Wrote data to /home/hojelse/repo/git-truck/parser/home/hojelse/data.json [2ms]
```

### LINKED ISSUE #334: Pass arguments to parser instead of frontend

### PR DESCRIPTION
**Pull request by hojelse**
Resolves #334
Fixes #333

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -6,11 +6,13 @@
...
-    "dev": "npm run parse && npx cross-env NODE_ENV=development remix dev",
+    "dev": "npm run parse --",
+    "postdev": "npx cross-env NODE_ENV=development remix dev",
     "postinstall": "remix setup node",
-    "start": "npm run parse && npx cross-env NODE_ENV=production remix-serve build",
+    "start": "npm run parse --",
+    "poststart": "npx cross-env NODE_ENV=production remix-serve build",
     "clean": "npx rimraf -rf ./parser/dist data.json build",
-    "parse": "cd parser && npm install && npm run build && node . --path .. --out ../data.json && cd .."
+    "parse": "npm --prefix ./parser/ install && npm --prefix ./parser/ run build && npm --prefix ./parser/ start -- --gittruckroot .. --path . --out ./data.json"
```
Comment (hojelse) [inline package.json:15]: Maybe it's little hacky to have the parser as the main script and frontend as a post- script?

File: package.json [hunk 2]
```diff
@@ -6,11 +6,13 @@
...
   },
```
Comment (hojelse) [inline package.json:16]: Is this a hacky solution? #330

File: parser/src/parse.ts [hunk 3]
```diff
@@ -205,7 +206,12 @@ export async function parse(rawArgs: string[]) {
     process.exit(1)
   }
 
-  const repoDir = args.path ?? "."
+  const truckroot_or_cwd = args.gittruckroot ?? process.cwd()
```
Comment (hojelse) [inline parser/src/parse.ts:209]: This is meant to allow for running the parser from`git-truck` and `git-truck/parser/` directly

File: package.json (hunk 2 above)
Comment (joglr) [reply package.json:16]: Maybe it makes more sense to have parse as a prestart script?

Comment (joglr) [reply package.json:16]: Ahh, that makes sense.

File: parser/src/parse.ts (hunk 3 above)
Comment (joglr) [reply parser/src/parse.ts:209]: Great change! 
Why the rename though? I don't really like the name `truckroot_or_cwd`

File: parser/src/parse.ts [hunk 4]
```diff
@@ -229,7 +234,11 @@ export async function parse(rawArgs: string[]) {
     "Commit tree hydrated",
     "Error hydrating commit tree"
   )
-  const outPath = join(process.cwd(), outFileName)
+
+  let outPath = args.out ?? `${repoName}_${branchName}.json`
+  if (!path.isAbsolute(outPath))
+    outPath = path.resolve(truckroot_or_cwd, outPath)
```
Comment (joglr) [inline parser/src/parse.ts:240]: Can you import resolve only from path?
So
import { resolve } from "path"

File: parser/src/parse.ts [hunk 5]
```diff
@@ -241,7 +250,7 @@ export async function parse(rawArgs: string[]) {
       }),
     "Writing data to file",
     `Wrote data to ${resolve(outPath)}`,
```
Comment (joglr) [inline parser/src/parse.ts:252]: We already use resolve as a named import here

File: parser/src/parse.ts [hunk 6]
```diff
@@ -22,6 +22,7 @@ import { resolve } from "path"
 import { performance } from "perf_hooks"
 import yargsParser from "yargs-parser"
 import { hydrateData } from "./hydrate"
+import path from "path/posix"
```
Comment (joglr) [inline parser/src/parse.ts:25]: is `path/posix` special?

File: parser/src/parse.ts (hunk 3 above)
Comment (joglr) [reply parser/src/parse.ts:209]: Can you maybe explain the reasoning for having two args for specifying which directory to analyze?

File: parser/src/parse.ts (hunk 6 above)
Comment (hojelse) [reply parser/src/parse.ts:25]: Not sure, this is the documentation: https://nodejs.org/api/path.html

File: parser/src/parse.ts (hunk 3 above)
Comment (hojelse) [reply parser/src/parse.ts:209]: - gittruckroot should always point to the git truck root directory. Only used when calling parser from the root npm script.
- repoDir, is the target path to analyse(just like before)

File: parser/src/parse.ts (hunk 5 above)
Comment (hojelse) [reply parser/src/parse.ts:252]: Oh then probably I should remove the extra import

File: parser/src/parse.ts (hunk 3 above)
Comment (joglr) [reply parser/src/parse.ts:209]: Okay, cool. Makes sense.

File: parser/src/parse.ts (hunk 6 above)
Comment (joglr) [reply parser/src/parse.ts:25]: I think the root path module is cross platform, while `path/posix` is not

**Pull request merged**